pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -215,6 +219,10 @@ func (cr *PerconaXtraDBCluster) Validate() error {
 		if err := c.ProxySQL.VolumeSpec.validate(); err != nil {
 			return errors.Wrap(err, "ProxySQL: validate volume spec")
 		}
+
+		if c.ProxySQL.Size < minSafeProxySQLSize && !c.AllowUnsafeConfig {
+			c.ProxySQL.Size = minSafeProxySQLSize
+		}
```

Comment:
I think `setSafeDefaults` function is better for setting new values. please excuse me for the confusion :(

Reply:
No problem. Fixed in 74953bd